src/app/functions/server/featureFlags.ts

```diff
@@ -7,10 +7,22 @@ import {
   FeatureFlag,
   getFeatureFlagByName,
 } from "../../../db/tables/featureFlags";
+import { getServerSession } from "next-auth";
+import { authOptions } from "../../api/utils/auth";
+
+export type FeatureFlagsEnabled = {
+  FreeBrokerScan: boolean;
+  PremiumBrokerRemoval: boolean;
+};
+
+export async function isFlagEnabled(name: string): Promise<boolean> {
+  const session = await getServerSession(authOptions);
```

Comment:
This might be more of a personal preference than a valid remark, so feel free to ignore, but I tend to try minimising (ideally to 0) the number of side effects in my function calls. So it might be preferable to just have the caller pass in a `user` object?

- In most use cases, I expect that the caller will have access to that already, so it would save an additional unwrapping of the JWT.
- It would remove the need to mock the module in the test.
- It would allow checking if some _other_ user is able to see something (possibly useful in an admin page, or if the user has just logged out).
- We can do away with the runtime error by making it a required parameter. (Or actually make it optional, for flags without an allowlist.)

Reply:
That's a good idea, thanks! I'd prefer to have more side-effect-free functions too.

Reply:
OK done - left the mock in place for the DB call, but that seems reasonable in this case IMHO and is fairly painless to mock, but we could consider if there's a better way.